src/masternodes/mn_checks.cpp

```diff
@@ -4348,6 +4393,105 @@ bool IsVaultPriceValid(CCustomCSView& mnview, const CVaultId& vaultId, uint32_t
     return true;
 }
 
+bool IsPaybackWithCollateral(CCustomCSView& view, const std::map<DCT_ID, CBalances>& loans) {
+    auto tokenDUSD = view.GetToken("DUSD");
+    if (!tokenDUSD)
+        return false;
+
+    if (loans.size() == 1
+    && loans.count(tokenDUSD->first)
+    && loans.at(tokenDUSD->first) == CBalances{{{tokenDUSD->first, 999999999999999999LL}}}) {
```

Comment:
is this ever true? looks weird.

Reply:
Yes, this allows two ways to payback with collateral. One with the new `paybackwithcollateral` RPC and one with `paybackloan` RPC and "9999999999.99999999@DUSD" passed as amounts.
This eases and speeds up the integration by downstream users by bypassing the need to support a new RPC and TX message.

Reply:
how can I resolve discussions? do I not have the rights or am I not finding the button? 🤔

Reply:
Must be a right issue. Let me resolve it.